server/athenian/api/controllers/settings_controller.py

```diff
@@ -566,26 +579,45 @@ async def clean_repository_sets():
             model.acc_id == account,
             model.repository_full_name == full_name,
         ))))
+    clean_deployments_coro = _clean_logical_deployments(
+        [full_name, drop_logical_repo(full_name)], account, pdb,
+    )
+    await gather(*tasks, clean_deployments_coro, op="_delete_logical_repository/sql")
+
+
+async def _clean_logical_deployments(
+    repos: Sequence[str], account: int, pdb: DatabaseLike, op: str = None,
+) -> None:
+    """Delete all deployments info about `repos` from pdb."""
+    deployment_models = (
+        GitHubCommitDeployment, GitHubPullRequestDeployment, GitHubReleaseDeployment,
+    )
     affected_deployments = await pdb.fetch_all(union(*(
         select([distinct(model.deployment_name)]).where(and_(
             model.acc_id == account,
-            model.repository_full_name.in_([full_name, drop_logical_repo(full_name)]),
+            model.repository_full_name.in_(repos),
         ))
-        for model in (GitHubCommitDeployment, GitHubPullRequestDeployment, GitHubReleaseDeployment)
+        for model in deployment_models
     )))
+    tasks = []
     if affected_deployments:
-        for model in (GitHubCommitDeployment,
-                      GitHubPullRequestDeployment,
-                      GitHubReleaseDeployment):
-            tasks.append(delete(model).where(and_(
-                model.acc_id == account,
-                model.repository_full_name.in_([full_name, drop_logical_repo(full_name)]),
-            )))
-        tasks.append(delete(GitHubDeploymentFacts).where(and_(
-            GitHubDeploymentFacts.acc_id == account,
-            GitHubDeploymentFacts.deployment_name.in_(affected_deployments),
-        )))
-    await gather(*tasks, op="_delete_logical_repository/sql")
+        log = logging.getLogger(f"{metadata.__package__}._clean_logical_deployments")
+        log.info(
+            "Cleaning deployments %s for repos %s", [r[0] for r in affected_deployments], repos,
```

Comment:
We should save `[r[0] for r in affected_deployments]` to a variable, we use it two times.

Reply:
done